File: src/main/java/com/zhan_dui/data/AppDatabase.java
package com.zhan_dui.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {MemoEntity.class}, version = 1, exportSchema = false)
@TypeConverters({}) // No converters needed for now
public abstract class AppDatabase extends RoomDatabase {
    private static volatile AppDatabase INSTANCE;

    public abstract MemoDao memoDao();

    public static AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                                    AppDatabase.class, "EverMemo.db")
                            .fallbackToDestructiveMigration() // Since we have no migration yet
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/zhan_dui/data/AppDatabase.kt
package com.zhan_dui.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import kotlin.concurrent.Volatile

@Database(entities = [MemoEntity::class],
    version = 2,
    exportSchema = false)
//@TypeConverters([]) // No converters needed for now
abstract class AppDatabase : RoomDatabase() {
    abstract fun memoDao(): MemoDao

    companion object {
        @Volatile
        private var instance: AppDatabase? = null

        @JvmStatic
        fun getInstance(context: Context): AppDatabase {
            return instance ?: synchronized(this) {
                instance ?: buildDatabase(context).also { instance = it }
            }
        }

        private fun buildDatabase(context: Context): AppDatabase {
            return Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "EverMemo.db"
            )
                .fallbackToDestructiveMigration() // Since we have no migration yet
                .build()

        }
    }
}